File: mongo-alpakka/src/main/scala/com/stulsoft/pmongo/alpakka/data/Test01Manager.scala
/*
 * Copyright (c) 2020. Yuriy Stul
 */

package com.stulsoft.pmongo.alpakka.data

import akka.{Done, NotUsed}
import akka.stream.Materializer
import akka.stream.alpakka.mongodb.scaladsl.MongoSource
import akka.stream.scaladsl.{Sink, Source}
import com.mongodb.reactivestreams.client.{MongoClient, MongoClients, MongoCollection, MongoDatabase}
import com.stulsoft.pmongo.alpakka.Config
import com.typesafe.scalalogging.StrictLogging
import org.bson.codecs.configuration.CodecRegistries.{fromProviders, fromRegistries}
import org.bson.codecs.configuration.CodecRegistry
import org.mongodb.scala.MongoClient.DEFAULT_CODEC_REGISTRY
import org.mongodb.scala.bson.codecs.Macros._

import scala.concurrent.duration.DurationInt
import scala.concurrent.{Await, Future}

/**
 * @author Yuriy Stul
 */
object Test01Manager extends StrictLogging {
  val codecRegistry: CodecRegistry = fromRegistries(fromProviders(classOf[Test01]), DEFAULT_CODEC_REGISTRY)
  val client: MongoClient = MongoClients.create(Config.mongoConnectionString())
  val db: MongoDatabase = client.getDatabase("pmongo")

  val test01Collection: MongoCollection[Test01] = db
    .getCollection("test_01", classOf[Test01])
    .withCodecRegistry(codecRegistry)

  def showAllDocuments()(implicit mat: Materializer): Unit = {
    logger.info("==>showAllDocuments")
    val source: Source[Test01, NotUsed] =
      MongoSource(test01Collection.find(classOf[Test01]))

    val rows: Future[Seq[Test01]] = source.runWith(Sink.seq)
    Await.result(rows, 5.seconds).foreach(t => logger.info("{}", t))

  }

  def listAllDocuments()(implicit mat: Materializer): Seq[Test01] = {
    logger.info("==>listAllDocuments")
    val source: Source[Test01, NotUsed] =
      MongoSource(test01Collection.find(classOf[Test01]))

    val rows: Future[Seq[Test01]] = source.runWith(Sink.seq)
    Await.result(rows, 5.seconds)
  }

  def showAllDocuments2()(implicit mat: Materializer): Future[Done] ={
    logger.info("==>showAllDocuments2")

    val source: Source[Test01, NotUsed] =
      MongoSource(test01Collection.find(classOf[Test01]))

    source.runWith(Sink.foreach(t => logger.info("{}", t)))
  }
}

File: mongo-alpakka-kotlin/src/main/kotlin/com/stulsoft/pmongo/alpakka/kotlin/Test01Manager.kt
/*
 * Copyright (c) 2020. Yuriy Stul
 */

package com.stulsoft.pmongo.alpakka.kotlin

import akka.NotUsed
import akka.actor.ActorSystem
import akka.stream.alpakka.mongodb.javadsl.MongoSource
import akka.stream.javadsl.Sink
import akka.stream.javadsl.Source
import com.mongodb.reactivestreams.client.MongoClient
import com.mongodb.reactivestreams.client.MongoClients
import com.mongodb.reactivestreams.client.MongoDatabase
import org.bson.Document


/**
 * @author Yuriy Stul
 */
object Test01Manager {
    private val client: MongoClient = MongoClients.create(Config.mongoConnectionString())
    private val db: MongoDatabase = client.getDatabase("pmongo")
    private val test01Collection = db.getCollection("test_01")

    fun showAllDocuments(system: ActorSystem) {
        val source: Source<Document, NotUsed> =
                MongoSource.create(test01Collection.find())
        source.runWith(Sink.foreach { d -> println(d) }, system)
    }
}
